app/modules/tunnel/socket_server.py

```
@@ -0,0 +1,456 @@
+
+async def _verify_token(token: Optional[str]) -> Optional[dict]:
+    """Verify Firebase ID token; return decoded claims or None."""
+    if not token or not isinstance(token, str):
+        return None
+    if os.getenv("isDevelopmentMode") == "enabled":
+        return {"user_id": os.getenv("defaultUsername", "dev-user"), "email": "dev@potpie.ai"}
+    try:
+        decoded = firebase_auth.verify_id_token(token)
+        if "user_id" not in decoded and "uid" in decoded:
+            decoded["user_id"] = decoded["uid"]
+        return decoded
```

Ruff (0.15.1)

[warning] Consider moving this statement to an `else` block

(TRY300)